src/main/java/Tasks.java

```diff
@@ -0,0 +1,173 @@
+import java.io.*;
+import java.util.ArrayList;
+
+public class Tasks implements Serializable {
+    private static ArrayList<Task> tasks;
+    private static final String FILE_PATH = "src/data/taskList.txt";
+    public Tasks() {
+        // Load tasks from file on startup
+        loadTasks();
+    }
+
+    private static void loadTasks() {
+        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(FILE_PATH))) {
+            tasks = (ArrayList<Task>) ois.readObject();
+        } catch (IOException | ClassNotFoundException e) {
+            // Handle file not found or corrupted data file
+            //System.err.println("Error loading tasks: " + e.getMessage());
+            tasks = new ArrayList<>();
+        }
+    }
+
+    public static void saveTask()  {
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(FILE_PATH))) {
+            oos.writeObject(tasks);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void markTask(String command) throws DukeException {
+        String[] words = command.split(" ");
+        if (words.length == 2) {
+            int taskId = Integer.parseInt(words[1]) -1;
+            if(taskId < tasks.size()) {
+                Task t = tasks.get(taskId);
+                t.markAsDone();
+                System.out.println(" Nice ! I've marked this task as done: \n" + t);
+            } else {
+                throw new DukeException(" Invalid task index. Type 'list' to list out your tasks. ");
+            }
+        } else {
+            throw new DukeException(" Invalid format. Type 'mark 1' to mark task 1. ");
+        }
```

Comment:
Might want to avoid deep nesting 😄 

By using guard clauses you can do stuff like:

```java
if (words.length != 2) {
    throw new ...
}
// rest of your code
...
```

The deeper the nesting, the harder it is for the keep track of the logic!

Deep nesting: ℹ️ 
https://nus-cs2103-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting